File: CSharp/ImageProcessing/ComSaveFileDialog.cs
﻿using Microsoft.Win32;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ImageProcessing
{
    class ComSaveFileDialog
    {
        protected SaveFileDialog m_saveFileDialog;

        public String FileName
        {
            set { m_saveFileDialog.FileName = value; }
            get { return m_saveFileDialog.FileName; }
        }
        public String InitialDirectory
        {
            set { m_saveFileDialog.InitialDirectory = value; }
            get { return m_saveFileDialog.InitialDirectory; }
        }

        public String Filter
        {
            set { m_saveFileDialog.Filter = value; }
            get { return m_saveFileDialog.Filter; }
        }

        public int FilterIndex
        {
            set { m_saveFileDialog.FilterIndex = value; }
            get { return m_saveFileDialog.FilterIndex; }
        }

        public String Title
        {
            set { m_saveFileDialog.Title = value; }
            get { return m_saveFileDialog.Title; }
        }

        public bool CheckFileExists
        {
            set { m_saveFileDialog.CheckFileExists = value; }
            get { return m_saveFileDialog.CheckFileExists; }
        }

        public bool CheckPathExists
        {
            set { m_saveFileDialog.CheckPathExists = value; }
            get { return m_saveFileDialog.CheckPathExists; }
        }

        public ComSaveFileDialog() : base()
        {
            m_saveFileDialog = new SaveFileDialog();
        }

        ~ComSaveFileDialog()
        {
        }

        public bool ShowDialog()
        {
            bool bRst = false;

            if (m_saveFileDialog.ShowDialog() == true)
            {
                bRst = true;
            }

            return bRst;
        }

        public void StreamWrite(string _str)
        {
            Stream stream = m_saveFileDialog.OpenFile();
            StreamWriter streamWriter = new StreamWriter(stream, Encoding.GetEncoding("UTF-8"));
            streamWriter.Write(_str);
            streamWriter.Close();
            stream.Close();

            return;
        }
    }
}

File: Vb/ImageProcessing/Lib/ComSaveFileDialog.vb
﻿Imports System.IO
Imports System.Text
Imports Microsoft.Win32

Namespace ImageProcessing
    Public Class ComSaveFileDialog
        Protected m_saveFileDialog As SaveFileDialog

        Public Property FileName() As String
            Set(value As String)
                m_saveFileDialog.FileName = value
            End Set
            Get
                Return m_saveFileDialog.FileName
            End Get
        End Property

        Public Property InitialDirectory() As String
            Set(value As String)
                m_saveFileDialog.InitialDirectory = value
            End Set
            Get
                Return m_saveFileDialog.InitialDirectory
            End Get
        End Property

        Public Property Filter() As String
            Set(value As String)
                m_saveFileDialog.Filter = value
            End Set
            Get
                Return m_saveFileDialog.Filter
            End Get
        End Property

        Public Property FilterIndex() As Integer
            Set(value As Integer)
                m_saveFileDialog.FilterIndex = value
            End Set
            Get
                Return m_saveFileDialog.FilterIndex
            End Get
        End Property

        Public Property Title() As String
            Set(value As String)
                m_saveFileDialog.Title = value
            End Set
            Get
                Return m_saveFileDialog.Title
            End Get
        End Property

        Public Property CheckFileExists() As Boolean
            Set(value As Boolean)
                m_saveFileDialog.CheckFileExists = value
            End Set
            Get
                Return m_saveFileDialog.CheckFileExists
            End Get
        End Property

        Public Property CheckPathExists() As Boolean
            Set(value As Boolean)
                m_saveFileDialog.CheckPathExists = value
            End Set
            Get
                Return m_saveFileDialog.CheckPathExists
            End Get
        End Property

        Public Sub New()
            m_saveFileDialog = New SaveFileDialog()
        End Sub

        Protected Overloads Sub Finalize()
            MyBase.Finalize()
        End Sub

        Public Function ShowDialog() As Boolean
            Dim bRst As Boolean = False

            If (m_saveFileDialog.ShowDialog() = True) Then
                bRst = True
            End If

            Return bRst
        End Function

        Public Sub SreamWrite(_str As String)
            Dim stream As Stream = m_saveFileDialog.OpenFile()
            Dim streamWriter As StreamWriter = New StreamWriter(stream, Encoding.GetEncoding("UTF-8"))
            streamWriter.Write(_str)
            streamWriter.Close()
            stream.Close()

            Return
        End Sub
    End Class
End Namespace